include/nighthawk/client/header_generator.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <functional>
+
+#include "envoy/http/header_map.h"
+
+namespace Nighthawk {
+namespace Client {
+
+using HeaderMapPtr = std::shared_ptr<const Envoy::Http::HeaderMap>;
+using GeneratorSignature = std::function<HeaderMapPtr()>;
+
+class HeaderGenerator {
```

Comment:
Technically this is a HeaderGeneratorGenerator since it returns (repeatedly) generators for headers.   However that is unwieldy.   This also makes GeneratorSignature a bit confusing.    Also that is a concrete class and technically not a signature (which only includes the input and output in an abstract way).

Perhaps s/GeneratorSignature/HeaderGenerator/ and then the class might be something like HeaderSource or somesuch.  I am not sure really, but I did find the levels of generator confusing.  HeaderStream perhaps ?  Other ideas?

Reply:
Oh, yeah the naming, I wanted to do something about that myself as well,  but forgot
 +1 to `s/GeneratorSignature/HeaderGenerator/` and `s/HeaderGenerator/HeaderSource/`. 
Sorry about that, it's quite confusing right now.

Reply:
I renamed this accordingly, and also relocated `header_source.*` from `client/` to `common/`